### PR TITLE
Added new GH Action to build website

### PR DESCRIPTION
**Pull request by rashidnhm**
This PR adds a new GH Actions to build and push the pennylane.ai site to the new S3 buckets

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          python compile.py
+
+      - name: Upload Pull Request Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: site
+          path: site
+          retention-days: ${{ github.event_name == 'pull_request' && 10 || 30 }}
+
+      - name: Push changes to dev (pennylane-ai-website-website-non-react-dev)
```
Comment (smokingroosters) [inline .github/workflows/build.yml:45]:
```suggestion
      - name: Push changes to dev (pennylane-ai-website-non-react-dev)
```
not sure if it's a typo? also repeated below if that's the case

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          AWS_ACCESS_KEY_ID: ${{ secrets.PL_SITE_DEV_NON_REACT_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.PL_SITE_DEV_NON_REACT_SECRET_ACCESS_KEY }}
+        run: aws s3 sync site s3://pennylane-ai-website-website-non-react-dev
+
+      - name: Push changes to prod (pennylane-ai-website-non-react-website-prod)
+        if: github.event_name == 'push' && github.ref_name == 'master'
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.PL_SITE_PROD_NON_REACT_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.PL_SITE_PROD_NON_REACT_SECRET_ACCESS_KEY }}
+        run: aws s3 sync site s3://pennylane-ai-website-non-react-website-prod
```
Comment (smokingroosters) [inline .github/workflows/build.yml:57]: I guess `non-react` name is only temporary until we have both active?

File: .github/workflows/build.yml (hunk 1 above)
Comment (rashidnhm) [reply .github/workflows/build.yml:45]: :sweat_smile: ... I accidentally added -website- twice to the S3 bucket name .. this is the name of S3 bucket verbatim ... I can change it but changing project name will require re-deploy of all infra

Comment (rashidnhm) [reply .github/workflows/build.yml:45]: Now that I'm thinking about it, I can potentially create a new bucket without the repeat and do an S3 sync :thinking: ... let me fix this one sec

File: .github/workflows/build.yml (hunk 2 above)
Comment (rashidnhm) [reply .github/workflows/build.yml]: non-react is temporary until we have phase 3 release of the react website, at that point this bucket will be removed and all content will be served from `pennylane-ai-website-prod`.

File: .github/workflows/build.yml (hunk 1 above)
Comment (rashidnhm) [reply .github/workflows/build.yml:45]: Bucket names updated :+1:

**Review by smokingroosters — APPROVED** [review]
LGTM! Just that name that can now be changed I guess

**Pull request merged**